pacman/operations/algorithm_reports/routing_compression_checker_report.py

```diff
@@ -0,0 +1,161 @@
+from collections import OrderedDict
+import logging
+import os
+
+from spinn_utilities.progress_bar import ProgressBar
+from pacman.exceptions import PacmanRoutingException
+from pacman.operations.algorithm_reports import reports
+
+logger = logging.getLogger(__name__)
+
+
+WILDCARD = "*"
+LINE_FORMAT = "0x{:08X} 0x{:08X} 0x{:08X} {: <7s} {}\n"
+
+
+def codify(route, length=32):
```

Comment:
Does this function just return the string value of key & mask?

Reply:
Instead of working with continuously with lots of keys, masks and binary operations what this does is convert the key and mask into a single string with 1, 0 or *.

Such that is only a 1 is accepted add a 1, if only 0 add a zero and use * if both are accepted.

This just speeds up comparison between two Strings instead of comparing two sets of keys and masks.

Reply:
Or at least made is so I could understand what was going on!